src/common/IterationLookup.ts

```diff
@@ -0,0 +1,280 @@
+import { DateTime } from "luxon";
+
+/**
+ * Manual overrides for the iteration lookup below. This can be used to fix
+ * erroneous iteration numbers and date ranges on Bugzilla. It can't be used to
+ * add new iterations, since that would require sorting, which is slow.
+ * @example { iteration: "66.1", range: "Dec 10 - 23" }
+ * @example { iteration: "66.2", range: null } // use null to exclude iterations
+ */
+const ITERATION_OVERRIDES = [];
+
+export interface IterationLookup {
+  // iterations by date in YYYY-MM-DD format
+  byDate: { [date: string]: string };
+  // date info by iteration string e.g. "100.1"
+  byVersionString: {
+    [versionString: string]: {
+      startDate: string; // In DateTime ISO format with timezone
+      endDate: string;
+      weeks: number; // Number of Mondays spent in the iteration
+    };
+  };
+  // List of versions, ordered by iteration number
+  orderedVersionStrings: string[];
+}
+
+/**
+ * For a given list of iteration strings, make an object with lookup tables for
+ * iteration strings and dates. Each string must have an iteration number (e.g.
+ * 100.1) and a date range. Order is important so the date computations work
+ * correctly, and so we're able to parse duplicates as overrides.
+ * @example $ new IterationLookup(["101.1 - Jan 4 - 15", "101.2 - Jan 18 - 29"])
+ *          > {
+ *              byDate: { "2019-01-07": "101.1", "2019-01-14": "101.1", ... },
+ *              byVersionString: { "101.1": { startDate: "2019-01-04", ... } },
+ *              orderedVersionStrings: ["101.1", "101.2"], ...
+ *            }
+ */
+export class IterationLookup {
```

Comment:
This is a class with no methods, only a ctor, which is a code smell.

This should be a function that returns a data structure.

Also having an interface `IterationLookup` and `class IterationLookup` is confusing from a typing perspective.

IMO rename this something like `buildIterationLookup` or `generateIterationLookup` or `lookupIterations.` and make it a function that returns an `IterationLookup` record.

Reply:
Thanks! Will fix these issues.